features/keychain/module/crypto/tweak.js

```diff
@@ -0,0 +1,24 @@
+import ecc from '@exodus/bitcoinerlab-secp256k1'
+import { utils } from '@noble/secp256k1'
+
+const tapTweakHash = (publicKey, h) => {
+  const xOnlyPoint = ecc.xOnlyPointFromPoint(publicKey)
+  const hash = utils.taggedHashSync('TapTweak', Buffer.concat(h ? [xOnlyPoint, h] : [xOnlyPoint]))
+  return Buffer.from(hash)
+}
+
+export const tweakPrivateKey = ({ hdkey, tweakOptions = {} }) => {
+  const { publicKey } = hdkey
+  const { tweakHash } = tweakOptions
+
+  let privateKey = publicKey[0] === 3 ? ecc.privateNegate(hdkey.privateKey) : hdkey.privateKey
+  privateKey = Buffer.from(ecc.privateAdd(privateKey, tapTweakHash(publicKey, tweakHash)))
+
+  return privateKey
+}
+
+export const tweakPublicKey = ({ hdkey, tweakOptions }) => {
+  const privateKey = tweakPrivateKey({ hdkey, tweakOptions })
+
+  return Buffer.from(ecc.pointFromScalar(privateKey, true))
+}
```

Comment:
maybe better accept `privateKey` everywhere here instead of `hdkey`? Seems like we only use that one field and it'd be nice to limit our options

Reply:
We use `privateKey` and `publicKey`.

Reply:
It will make `secp256k1.js` messier if I need to pass both.